### ISSUE TITLE
Feature Request: Import Highlights

### ISSUE THREAD

**Issue by edr1412** [obsidianmd/obsidian-clipper#111]
Currently, Obsidian Web Clipper supports **exporting highlights** to a file. It would be great if it also allowed **importing highlights** across devices, so we could view highlights on other devices.

An automatic sync option would be even more helpful!

**Comment by mufumade** [comment]
Yes would be great to sync the highlights directly to Obsidian so the highlights are in the vault and are also visible in all browsers.

**Comment by rawexx** [comment]
Браузер стал важной частью жизни каждого человека, сохраняя видимость всех основных моментов на основе хранилища обсидиана с любым устройством, профилем, браузером это будет совершенно новый уровень взаимодействия браузера и обсидиана, который сделает жизнь лучше для каждого пользователя обсидиана.
Извините за мой английский, я использую перевод

Это создает следующие проблемы:
1.1 переустановка ОС уничтожит все подсветки в браузере, и теперь ее невозможно восстановить
1.2 ошибка браузера уничтожит все подсветки в браузере, и теперь ее невозможно восстановить
1.3 ошибка операционной системы уничтожит все подсветки в браузере, и теперь ее невозможно восстановить
1.4 нулевая поддержка мультибраузерности И мультибраузер-профиля И мультиустройства
1.5 это неудобно, непрактично, недобросовестно, опасно
1.6 Создание дубликатов заметок для 1 страницы/сайта
1.7 Нет понимания, что на этом сайте уже есть заметка, когда вы на нем находитесь, если у вас нет хайлайта, который вы видели визуально

Лучший, +- легкий/средний уровень сложности, и возможные решения быстро выпускаются при использовании плагина [Obsidan REST API ](https://github.com/coddingtonbear/obsidian-local-rest-api)или добавляют некоторый функционал в Obsidian из плагина (лицензия MIT эта лицензия позволяет это сделать) с помощью dataviewJS, и, возможно, добавляют [бета-версию datacore](https://github.com/blacksmithgu/datacore) для 10-100-кратного увеличения скорости
[ALL Документация](https://coddingtonbear.github.io/obsidian-local-rest-api/)
API 1 Реализация (со ссылкой на документацию):
1.1 GET [/commands/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Commands/get_commands_) Получение списка доступных команд
1.2 POST [/commands/{commandId}/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Commands/post_commands__commandId__) Выполнение команды
1.3 POST [/search/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Search/post_search_) Поиск документов, соответствующих указанному поисковому запросу и представлению
данных 1.4 POST [/search/simple/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Search/post_search_simple_) Поиск документов, соответствующих указанному текстовому запросу
1.5 GET [/vault/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Directories/get_vault_) Список файлов, которые существуют в корне вашего хранилища
1.6 GET [/vault/{pathToDirectory}/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Directories/get_vault__pathToDirectory__) Список файлов, которые существуют в указанном каталоге
1.7 GET [/vault/{имя_файла}](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Files/get_vault__filename_) Возврат содержимого одного файла в вашем хранилище
1.8 PATCH [/vault/{имя_файла}](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Files/patch_vault__filename_) Вставка содержимого в существующую заметку относительно заголовка в этом документе.
1.9 PUT [/vault/{filename}](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Files/put_vault__filename_) Создайте новый файл в вашем хранилище или обновите содержимое существующего.

2 ДОБАВИТЬ возможность вызова новых функций из шаблонов
плагинов 3 для очень быстрого запроса по умолчанию запрос (без dataviewJS) Создать индекс с UUID И все выделения И все заметки с этим выделением И со всеми ссылками на веб-сайт, где были созданы основные моменты И когда создали, И с какого профиля\браузера был сделан этот пост (ДОБАВИТЬ поле для ввода в настройках плагина (глобального и шаблонного) "как называется этот браузер\профиль браузера" опция по умолчанию = obsidian веб)
4. UUID добавить к заметке в метаданные
5 Для продвинутых запросов использовать dataview И ВОЗМОЖНО datacore + omnisearch
6 возможность использовать команды после ИЛИ перед сохранением заметки
7 Настройка для шаблонов: ЕСЛИ ссылка на сайт есть в метаданных используйте этот шаблон, если в теле заметки, но нет в метаданных этот шаблон (применить шаблон 2)
8 Когда есть заметка с сайтом, отобразите иконку на вкладке иконка расширения и всплывающее окно (опционально)
![Uploading image.png…]()
Дополнительные опции для лучшего опыта
Добавить заметку как заметку с активированным шаблоном (для запуска скрипта всплывающее окно для данных поля или пользовательский js для работы с данными)
Это то же самое, но для QuickAdd

**Comment by rawexx** [comment]
Best, +- easy/medium level difficulty, and possible solutions to the fix problem and fast to release when using plugin [Obsidan REST API ](https://github.com/coddingtonbear/obsidian-local-rest-api)OR add some functional to obsidian from plugin (MIT License this license allows to do that) @coddingtonbear with dataviewJS, and maybe add [beta datacore](https://github.com/blacksmithgu/datacore) for 10-100x better speed
[ALL API Documentation](https://coddingtonbear.github.io/obsidian-local-rest-api/)
1 Implementation (with link to documentation):
1.1 GET [/commands/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Commands/get_commands_) Get a list of available commands
1.2 POST [/commands/{commandId}/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Commands/post_commands__commandId__) Execute a command
1.3 POST [/search/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Search/post_search_) Search for documents matching a specified search query and dataview
1.4 POST [/search/simple/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Search/post_search_simple_) Search for documents matching a specified text query
1.5 GET [/vault/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Directories/get_vault_) List files that exist in the root of your vault
1.6 GET [/vault/{pathToDirectory}/](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Directories/get_vault__pathToDirectory__) List files that exist in the specified directory
1.7 GET [/vault/{filename}](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Files/get_vault__filename_) Return the content of a single file in your vault
1.8 PATCH [/vault/{filename}](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Files/patch_vault__filename_) Insert content into an existing note relative to a heading within that document.
1.9 PUT [/vault/{filename}](https://coddingtonbear.github.io/obsidian-local-rest-api/#/Vault%20Files/put_vault__filename_) Create a new file in your vault or update the content of an existing one.

2 ADD the ability to call new functions from plugin templates
3 for very fast query default query(without dataviewJS) Create index with UUID AND all highlights AND all notes with this highlights AND with all website link where created highlights AND when created AND from which profile\browser this post was made (ADD field to enter in settings plugin (global and template) “what is the name of this browser\browser profile" default option = obsidian web)
4. UUID add to note in metadata
5 For advanced queries, use dataview AND MAYBE datacore + omnisearch
6 the ability to use commands after OR before saving a note
7 Customization for templates: IF the link to the site is in the metadata use this template, if in the body of the note, but not in the metadata this template (apply template 2)
8 When there is a note with website display the icon on the extension icon and popup (optional)
![image](https://github.com/user-attachments/assets/0dc150a3-b15a-4eeb-9b2e-079b99eae1dc)
Reference https://chromewebstore.google.com/detail/edoacekkjanmingkbkgjndndibhkegad

Additional Options for better experience
Add note as note with templater activated (for running script pop up for field data or custom js to working with data)
This same but for QuickAdd

More information here https://github.com/obsidianmd/obsidian-clipper/issues/317
